### PR TITLE
NuttX:new support for NuttX platform

### PR DESCRIPTION
**Pull request by crafcat7**
NuttX is a real-time operating system (RTOS) with an emphasis on standards compliance and small footprint. Scalable from 8-bit to 64-bit microcontroller environments, the primary governing standards in NuttX are Posix and ANSI standards. Additional standard APIs from Unix and other common RTOS’s (such as VxWorks) are adopted for functionality not available under these standards, or for functionality that is not appropriate for deeply-embedded environments (such as fork()).
NuttX is therefore compliant with the rtosbenchmark test platform
In that change, the following sections were added
- src/nuttx/bench_porting_layer_nuttx.c
- src/nuttx/bench_porting_layer_nuttx.h
- src/nuttx/timer/timer.c
and in
- h/bench_api.
- h/bench_utils.h
into src/nuttx/bench_porting_layer_nuttx.h

### CODE REVIEW

**Review by peter-mitsis — CHANGES_REQUESTED** [review]
Adding support for comparing against another RTOS--fantastic!

I have one request before giving it the approval for rebase/merge. As this benchmark suite is a sub-project of Zephyr, I think it would worthwhile to follow its coding style (see https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style). How does that apply here?  Please use tabs for indentation instead spaces, where 1 tab is equivalent to 8 spaces.

**Comment by crafcat7** [comment]
I am glad to receive your suggestion, on top of this I have made certain code formatting changes with the code,  thank you!

**Comment by peter-mitsis** [comment]
Almost there.

My apologies for not noticing this the first time.  Commit entries should be signed off. See https://docs.zephyrproject.org/latest/contribute/guidelines.html#developer-certification-of-origin-dco .

Second, the merge commit is not required, and should be deleted. We do our best to maintain a linear git history, so we tend to use "git rebase". The basic work flow is ...

```
git checkout main
git pull
git rebase -i main <your-branch>   // -i = interactive mode (allows for squashing, editing and dropping of commits)
git push -f origin <your-branch>   // -f = force the push (will replace the old branch if it exists)
```

**Comment by crafcat7** [comment]
Thank you for your suggestions, changes have been made.

**Review by peter-mitsis — APPROVED** [review]

**Pull request merged**
